app/models/ems_refresh/save_inventory_helper.rb

```diff
@@ -94,12 +98,17 @@ def save_inventory_with_findkey(association, hash, deletes, new_records, record_
       found = association.build(hash.except(:id))
       new_records << found
     else
-      found.update_attributes!(hash.except(:id, :type))
+      update_attributes!(found, hash, [:id, :type])
       deletes.delete(found) unless deletes.blank?
     end
     found
   end
 
+  def update_attributes!(ar_model, attributes, remove_keys)
+    ar_model.assign_attributes(attributes.except(*remove_keys))
+    ar_model.save! if ar_model.changed?
```

Comment:
Can you put a note here with a reference to the upstream issue?  Maybe something like

```ruby
# HACK: Avoid empty BEGIN/COMMIT pair until fix is made for https://github.com/rails/rails/issues/17937
```